subsys/bluetooth/mesh/gen_ponoff_srv.c

```diff
@@ -93,20 +93,19 @@ static void store_state(struct bt_mesh_ponoff_srv *srv)
 #endif
 }
 
-static void send_rsp(struct bt_mesh_ponoff_srv *srv,
-		     struct bt_mesh_msg_ctx *ctx)
+static void send_rsp(struct bt_mesh_ponoff_srv *srv, struct bt_mesh_msg_ctx *ctx, bool is_setup)
```

Comment:
Instead of adding `is_setup`, would suggest to add `on_power_up` argument:
```suggestion
static void send_rsp(struct bt_mesh_model *model, struct bt_mesh_msg_ctx *ctx, bool on_power_up)
```
Then, when `send_rsp` call will look a bit more clear:
- for Setup server: `send_rsp(srv->ponoff_setup_model, ctx, srv->on_power_up)`;
- for Server: `send_rsp(srv->ponoff_model, ctx, srv->on_power_up)`;

Reply:
I guess `is_setup` (or `use_setup` or just `setupctx`) would reflect nicely for all models, otherwise one will have to invent new names?

Reply:
I guess the point here is that you have the specific model and the message arguments as input into the function instead of using the boolean flag.